src/Tools/GenerateDocumentationAndConfigFiles/Program.cs

```diff
@@ -108,6 +108,11 @@ public static int Main(string[] args)
 
                     foreach (var rule in analyzer.SupportedDiagnostics)
                     {
+                        if (rule.Category == "Globalization")
+                        {
+                            Debug.Assert(analyzerType.BaseType?.Name == "AbstractGlobalizationDiagnosticAnalyzer");
+                        }
```

Comment:
This doesn't seem to be the correct/ideal place for this assertion. It makes more sense if this is turned into a unit test. Placing it here temporarily.